src/Data/Tree/AVL.agda

```diff
@@ -92,13 +92,13 @@ module _ {v} {V : Value v} where
 
   headTail : Tree V → Maybe (K& V × Tree V)
   headTail (tree (Indexed.leaf _)) = nothing
-  headTail (tree {h = suc _} t)    with Indexed.headTail t
-  ... | (k , _ , _ , t′) = just (k , tree (Indexed.castˡ ⊥⁺<[ _ ] t′))
+  headTail (tree {h = suc _} t) with (k , _ , _ , t′) ← Indexed.headTail t
+    = just (k , tree (Indexed.castˡ ⊥⁺<[ _ ] t′))
 
   initLast : Tree V → Maybe (Tree V × K& V)
   initLast (tree (Indexed.leaf _)) = nothing
-  initLast (tree {h = suc _} t)    with Indexed.initLast t
-  ... | (k , _ , _ , t′) = just (tree (Indexed.castʳ t′ [ _ ]<⊤⁺) , k)
+  initLast (tree {h = suc _} t) with (k , _ , _ , t′) ← Indexed.initLast t
```

Comment:
This can be turned into a `let`, surely?

Reply:
Yes, it can - but I'm not so sure that let is superior to irrefutable `with`. It makes no difference for complete, working code, but in a hole, things look quite different between the two. And let is bad in (current) Agda as it inlines eagerly (i.e. rather than saving on duplication, in Agda, it causes it!)

Reply:
Hopefully this will be fixed soon!?